src/main/java/com/nvidia/spark/rapids/jni/GpuTimeZoneDB.java

```diff
@@ -170,12 +174,56 @@ private void loadData(Executor executor) throws IllegalStateException {
     }
   }
 
+  /**
+   * load ZoneId.SHORT_IDS and append Z->UTC.
+   * The first 3 entries are: Z->UTC, PST->America/Los_Angeles, CTT->Asia/Shanghai
+   */
+  private void loadTimeZoneShortIDs() {
+    HostColumnVector.DataType type = new HostColumnVector.StructType(false,
+    new HostColumnVector.BasicType(false, DType.STRING),
+    new HostColumnVector.BasicType(false, DType.STRING));
+    ArrayList<HostColumnVector.StructData> data = new ArrayList<>();
+    // add Z->UTC
+    data.add(new HostColumnVector.StructData("Z", "UTC"));
+    // add PST CTT
+    for (Map.Entry<String, String> e : ZoneId.SHORT_IDS.entrySet()) {
+      if (e.getKey().equals("PST") || e.getKey().equals("CTT")) {
+        data.add(new HostColumnVector.StructData(e.getKey(), e.getValue()));
+      }
+    }
+    // add others
+    for (Map.Entry<String, String> e : ZoneId.SHORT_IDS.entrySet()) {
+      if (!(e.getKey().equals("PST") || e.getKey().equals("CTT"))) {
+        data.add(new HostColumnVector.StructData(e.getKey(), e.getValue()));
+      }
+    }
+    shortIDs = HostColumnVector.fromStructs(type, data);
+  }
+
+  public ColumnVector getTimeZoneShortIDs() {
+    return shortIDs.copyToDevice();
+  }
+
   @SuppressWarnings("unchecked")
   private void doLoadData() {
     synchronized (this) {
       try {
+        loadTimeZoneShortIDs();
         Map<String, Integer> zoneIdToTable = new HashMap<>();
         List<List<HostColumnVector.StructData>> masterTransitions = new ArrayList<>();
+        // Build a timezone ID index for the rendering of timezone IDs which may be included in datetime-like strings.
+        // For instance: "2023-11-5T03:04:55.1 Asia/Shanghai" -> This index helps to find the
+        // offset of "Asia/Shanghai" in timezoneDB.
+        //
+        // Currently, we do NOT support all timezone IDs. For unsupported ones, we ought to throw Exception anyway. And
+        // for invalid ones, we replace them with NULL value when ANSI mode is off. Therefore, we need to distinguish the
+        // unsupported ones from invalid ones which means the unsupported Ids need to be collected as well.
+        // To distinguish supported IDs from unsupported ones, we place all unsupported IDs behind supported ones:
+        // 1. Collect the IDs of all supported timezones in the order of masterTransitions.
+        // 2. Append the IDs of all unsupported timezones after the suported ones.
```

Comment:
Append them to what?

Reply:
Originally, this list first append supported time zones, then append unsupported time zones.
Becasue when parsing string to timestamp:
- If the time zone is not supported but it's valid in the string, should throw not supported exception regardless of ANSI mode.
- If the time zone is invalid, return null when ANSI mode is closed.
The transition table only contains supported time zone.
The kernel use time zone index in this list to distinguish supported/unsupported/invalid scenario.
- If index in [0, transition.size] it's supported
- If index in [transition.size, zondIdList.size] it's unsupported.

Will change to:
In order to fast the search, sort the zondIdList, use binary search to find the time zone index.
Because sort will break the supported and unsupported group

Reply:
Then should the list of string time zone names be a map instead? That way you look up the name using a binary search, and then get the int stored in the index to point to where the data actually is?

Reply:
Will update.